frontend/document/document.lua

```diff
@@ -173,8 +173,10 @@ end
 
 function Document:getProps(cached_doc_metadata)
     local function makeNilIfEmpty(str)
-        if str == "" then
+        if str == nil or str == "" then
             return nil
+        elseif str:sub(-1) == "\0" then
+            return str:sub(1, -2)
```

Comment:
This fill like papering over a bug that should be fixed where that string is created.

Reply:
https://github.com/koreader/koreader-base/blob/ecf6bc70c8f662ab00e819885b74a937eb59d332/ffi/mupdf.lua#L274-L287

Reply:
```c
/**
	Retrieve document meta data strings.

	doc: The document to query.

	key: Which meta data key to retrieve...

[…]

	buf: The buffer to hold the results (a nul-terminated UTF-8
	string).

	size: Size of 'buf'.

	Returns the number of bytes need to store the string plus terminator
	(will be larger than 'size' if the output was truncated), or -1 if the
	key is not recognized or found.
*/
int fz_lookup_metadata(fz_context *ctx, fz_document *doc, const char *key, char *buf, int size);
```

So can you test with this version:

```lua
local function getMetadataInfo(doc, info)
    local bufsize = 255
    local buf = ffi.new("char[?]", bufsize)
    local res = M.fz_lookup_metadata(context(), doc, info, buf, bufsize)
    if res > bufsize then
        bufsize = res
        buf = ffi.new("char[?]", bufsize)
        res = M.fz_lookup_metadata(context(), doc, info, buf, bufsize)
    end
    if res > 1 then
        return ffi.string(buf, res - 1)
    end
    return ""
end
```

Reply:
Thanks, that works.